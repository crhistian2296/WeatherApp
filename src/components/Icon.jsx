import PropTypes from 'prop-types';
import React from 'react';

const Icon = ({ iconId, weatherMain, size = '180px' }) => {
  return (
    <img
      loading='lazy'
      src={`https://openweathermap.org/img/wn/${iconId}@4x.png`}
      height={size}
      width={size}
      alt={`${weatherMain}`}
    />
  );
};

Icon.propTypes = {
  iconId: PropTypes.string.isRequired,
  weatherMain: PropTypes.string.isRequired,
  size: PropTypes.number,
};

export default Icon;
