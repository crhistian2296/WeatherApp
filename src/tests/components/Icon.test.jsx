import { render } from '@testing-library/react';
import React from 'react';
import Icon from '../../components/Icon';

describe('tests in <Icon/>', () => {
  test('should match its own snapshot', () => {
    const { container } = render(
      <Icon iconId={'sd1234'} weatherMain={'heavy rain'} />
    );
    expect(container).toMatchSnapshot();
  });
});
