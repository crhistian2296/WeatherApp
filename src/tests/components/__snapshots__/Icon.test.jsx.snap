// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <Icon/> should match its own snapshot 1`] = `
<div>
  <img
    alt="heavy rain"
    height="180px"
    loading="lazy"
    src="https://openweathermap.org/img/wn/sd1234@4x.png"
    width="180px"
  />
</div>
`;
